pkg/tfbridge/schema.go

```diff
@@ -1652,6 +1652,33 @@ func isDefaultOrZeroValue(tfs shim.Schema, ps *SchemaInfo, v resource.PropertyVa
 func extractSchemaInputs(state resource.PropertyValue, tfs shim.Schema, ps *SchemaInfo,
 	rawNames bool) resource.PropertyValue {
 
+	if ps == nil {
+		ps = &SchemaInfo{}
+	}
+
+	// If ps is MaxItemsOne, then it will not be present in the
+	for IsMaxItemsOne(tfs, ps) {
+		ps = ps.Elem
+		if ps == nil {
+			ps = &SchemaInfo{}
+		}
+
+		switch e := tfs.Elem().(type) {
+		case shim.Schema:
+			tfs = e
+		case shim.Resource:
+			if state.IsObject() {
+				return resource.NewProperty(
+					extractSchemaInputsObject(state.ObjectValue(), e.Schema(), ps.Fields),
+				)
+			} else {
+				// The shape of state and tfs have de-synchronized, so we
```

Comment:
Sure about this? This is tolerating schema mismatch? Tough choices.

Reply:
We were always tolerating schema mismatches, this just calls it out explicitly.

Reply:
Got it. Hmpf.